File: 6_ZigZag_Conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1 || numRows > s.size()) return s;
        
        vector<string> rows(numRows);
        
        for (int i = 0; i < s.size(); i++){
            int quo = (i / (numRows - 1)) % 2;
            int rem = i % (numRows - 1);
            int rowId = quo ? (numRows - 1 - rem) : rem;
            
            rows[rowId] += s[i];
        }
        
        string ret = rows[0];
        
        for (int j = 1; j < numRows; j++){
            ret = ret + rows[j];
        }
        
        return ret;
    }
};

File: 6_ZigZag_Conversion.py
class Solution:
    def convert(self, s: 'str', numRows: 'int') -> 'str':
        
        if numRows == 1 or numRows > len(s):
            return s
        
        rows= [''] * numRows
        
        for i in range (0, len(s)):
            quo = (i // (numRows-1)) % 2
            rem = i % (numRows - 1)
            if quo:
                rowId = numRows - 1 - rem
            else:
                rowId = rem
            
            rows[rowId] += s[i]
        
        ret = rows[0];
        
        for j in range(1, numRows):
            ret = ret + rows[j];
        
        return ret
